fv3core/stencils/updatedzd.py

```diff
@@ -17,132 +17,237 @@
+                ) / xt2
+                qe2x = (
+                    xt1 * q2x[0, 0, -1] + q2x[0, 0, -2] - a_bot * qe2x[0, 0, -1]
+                ) / xt2
+            with horizontal(region[:, local_js : local_je + 2]):
+                qe1y = (
+                    xt1 * q1y[0, 0, -1] + q1y[0, 0, -2] - a_bot * qe1y[0, 0, -1]
+                ) / xt2
+                qe2y = (
+                    xt1 * q2y[0, 0, -1] + q2y[0, 0, -2] - a_bot * qe2y[0, 0, -1]
+                ) / xt2
     with computation(BACKWARD), interval(0, -1):
-        qe1 = qe1 - gam * qe1[0, 0, 1]
-        qe2 = qe2 - gam * qe2[0, 0, 1]
-
-
-def edge_python(q1, q2, qe1, qe2, dp0, gam, islice, jslice, qe1_2, gam_2):
-    grid = spec.grid
-    dcol = dp0[0, 0, :]
-
-    km = grid.npz - 1
-    g0 = dcol[1] / dcol[0]
-    xt1 = 2.0 * g0 * (g0 + 1.0)
-    bet = g0 * (g0 + 0.5)
-
-    qe1[islice, jslice, 0] = (xt1 * q1[islice, jslice, 0] + q1[islice, jslice, 1]) / bet
-
-    qe2[islice, jslice, 0] = (xt1 * q2[islice, jslice, 0] + q2[islice, jslice, 1]) / bet
-    gam[islice, jslice, 0] = (1.0 + g0 * (g0 + 1.5)) / bet
-
-    for k in range(1, km + 1):
-        gk = dcol[k - 1] / dcol[k]
-        bet = 2.0 + 2.0 * gk - gam[islice, jslice, k - 1]
-        qe1[islice, jslice, k] = (
-            3.0 * (q1[islice, jslice, k - 1] + gk * q1[islice, jslice, k])
-            - qe1[islice, jslice, k - 1]
-        ) / bet
-        qe2[islice, jslice, k] = (
-            3.0 * (q2[islice, jslice, k - 1] + gk * q2[islice, jslice, k])
-            - qe2[islice, jslice, k - 1]
-        ) / bet
-        gam[islice, jslice, k] = gk / bet
-
-    a_bot = 1.0 + gk * (gk + 1.5)
-    xt1 = 2.0 * gk * (gk + 1.0)
-    xt2 = gk * (gk + 0.5) - a_bot * gam[islice, jslice, km]
-    qe1[islice, jslice, km + 1] = (
-        xt1 * q1[islice, jslice, km]
-        + q1[islice, jslice, km - 1]
-        - a_bot * qe1[islice, jslice, km]
-    ) / xt2
-    qe2[islice, jslice, km + 1] = (
-        xt1 * q2[islice, jslice, km]
-        + q2[islice, jslice, km - 1]
-        - a_bot * qe2[islice, jslice, km]
-    ) / xt2
-    for k in range(km, -1, -1):
-        qe1[islice, jslice, k] = (
-            qe1[islice, jslice, k] - gam[islice, jslice, k] * qe1[islice, jslice, k + 1]
-        )
-        qe2[islice, jslice, k] = (
-            qe2[islice, jslice, k] - gam[islice, jslice, k] * qe2[islice, jslice, k + 1]
-        )
+        qe1x, qe2x, qe1y, qe2y = edge_profile_reverse(qe1x, qe2x, qe1y, qe2y, gam)
+
+
+# def edge_python(q1, q2, qe1, qe2, dp0, gam, islice, jslice, qe1_2, gam_2):
+#     grid = spec.grid
+#     dcol = dp0[0, 0, :]
+#     km = grid.npz - 1
+#     g0 = dcol[1] / dcol[0]
+#     xt1 = 2.0 * g0 * (g0 + 1.0)
+#     bet = g0 * (g0 + 0.5)
+#     qe1[islice, jslice, 0] = (xt1 * q1[islice, jslice, 0] +
+#                                     q1[islice, jslice, 1]) / bet
+#     qe2[islice, jslice, 0] = (xt1 * q2[islice, jslice, 0] +
```

Comment:
Should we remove this code? in general I am not opposed to keeping it around for a bit, but is there any reason why it would be helpful to bring back?

Reply:
I would be happy to remove it.